orttraining/orttraining/core/session/training_session.cc

```diff
@@ -289,14 +354,28 @@ Status TrainingSession::ConfigureForTraining(
       weight_names_to_train, loss_name, config.gradient_graph_config, *session_logger_));
 
   if (config.pipeline_config.has_value()) {
+    // TODO: The number of pipeline steps should be passed in from Python API.
+    const int num_pipeline_steps = config.distributed_config.num_pipeline_steps;
+    const int num_pipeline_stages = config.distributed_config.pipeline_parallel_size;
+    pipeline_schedule_ = pipeline::PipelineScheduler(num_pipeline_steps, num_pipeline_stages);
+    pipeline_worker_pool_ = pipeline::PipelineWorkerPool(num_pipeline_stages);
+
+    // Insert InsertPipelineOps may access "sliced_schema" from "pipeline_context_".
```

Comment:
Typo, duplicate Insert

Reply:
Fixed.

---
In reply to: [527649515](https://github.com/microsoft/onnxruntime/pull/5815#discussion_r527649515) [](ancestors = 527649515)